app/connectors/PayApiConnector.scala

```diff
@@ -45,17 +47,45 @@ class PayApiConnector @Inject()(val http: HttpClient,
       }
   }
 
-  private def generateRequestForPayApi(balance: BigDecimal, sdilRef: String): SetupPayApiRequest = {
+  private def generateRequestForPayApi(balance: BigDecimal, sdilRef: String, optLastReturn: Option[SdilReturn],
```

Comment:
I'd put the default `todaysDate` param on here and not have it as a param at all on `initJourney` with a view towards mocking it if possible/time applicable

Reply:
I wanted it there, however, when I went to include tests for the expected logging, I needed the test to be initiated from initJourney.

Reply:
Ok. Makes sense.